dtfs-central-api/src/services/portal/deal.service.ts

```diff
@@ -40,4 +41,32 @@ export class PortalDealService {
       throw new Error(`Invalid dealType ${dealType}`);
     }
   }
+
+  /**
+   * Add a "deal cancelled" activity
+   *
+   * @param addDealCancelledActivity
+   * @param addDealCancelledActivity.dealId - the deal Id to update
+   * @param addDealCancelledActivity.dealType - the deal type
+   * @param addDealCancelledActivity.portalActivities - previous/existing deal activities
+   * @param addDealCancelledActivity.author - the activity's author
+   * @param addDealCancelledActivity.auditDetails - the users audit details
+   */
+  public static async addDealCancelledActivity({
+    dealId,
+    dealType,
+    portalActivities,
+    author,
+    auditDetails,
+  }: {
+    dealId: ObjectId | string;
+    dealType: DealType;
+    portalActivities: Array<Activity>;
+    author: ActivityAuthor;
+    auditDetails: AuditDetails;
+  }): Promise<void> {
+    if (dealType === DEAL_TYPE.GEF) {
+      await addGefDealCancelledActivity({ dealId, portalActivities, author, auditDetails });
+    }
+  }
```

Comment:
I think we should keep the conditional just incase it's called with a non GEF deal

Wanted to keep the actual activity logic in a separate function like `updateGefDealStatus()` and `updateBssEwcsDealStatus()` so that it can be used in other places.

However in hindsight this seems overkill now, so have moved the activity logic into the service 👍